rclcpp/src/rclcpp/generic_publisher.cpp

```diff
@@ -31,4 +31,47 @@ void GenericPublisher::publish(const rclcpp::SerializedMessage & message)
   }
 }
 
+void GenericPublisher::publish_loaned_msg(const rclcpp::SerializedMessage & message)
+{
+  if (this->can_loan_messages()) {
```

Comment:
I think that calling the `can_loan_messages` outside is the user's responsibility.
https://github.com/ros2/rclcpp/issues/1837#issuecomment-1006142083

Reply:
Yeah. Not call can_loan_messages() to check if this feature is enabled.   
Even if the user doesn't call can_loan_messages(), rcl function will return error code.